files/build_templates/docker_image_ctl.j2

```diff
@@ -153,6 +180,7 @@ start() {
 {%- endif %}
         --tmpfs /tmp \
         --tmpfs /var/tmp \
+        --hostname $HOSTNAME \
```

Comment:
>$HOSTNAME [](start = 19, length = 9)

Possible to input injection

Reply:
Still possible. For example:
`
sonichost"; rm -rf; echo "
`

---
In reply to: [286279985](https://github.com/Azure/sonic-buildimage/pull/2924#discussion_r286279985) [](ancestors = 286279985)

Reply:
Bash did not process metacharacters from variables only metacharacters directly present in a script or interpreter line.
It is easy to check by the next commands:
```bash
print_args() {
while (( "$#" )); do 
  echo $1 
  shift 
done
}
X="hostName"
Y='sonic"; echo injection; echo "'
print_args --hostname "$X" --name cname
#output
>>> --hostname
hostName
--name
cname

print_args --hostname "$Y" --name cname
#output
>>> --hostname
sonic"; echo injection; echo "
--name
cname
```

I thought that [CVE-2016-0634 ](https://www.cvedetails.com/cve/CVE-2016-0634/)can be exploited there. But sonic use bash 4.4

So, for any case added validation for allowed characters in a hostname.

Reply:
@qiluo-msft Could we close this conversation?

Reply:
Nice comments! Really appreciate the clarification and reference. 

---
In reply to: [286898693](https://github.com/Azure/sonic-buildimage/pull/2924#discussion_r286898693) [](ancestors = 286898693)